magefiles/magefile.go

```diff
@@ -123,14 +154,164 @@ func RunPreCommit() error {
 	return nil
 }
 
-// RunTests runs all of the unit tests
+// RunTests executes all unit and integration tests.
+//
+// **Returns:**
+//
+// error: An error if any issue occurs while running the tests.
 func RunTests() error {
 	mg.Deps(InstallDeps)
 
 	fmt.Println("Running unit tests.")
-	if err := sh.RunV(filepath.Join(".hooks", "run-go-tests.sh"), "all"); err != nil {
+	if _, err := sys.RunCommand(filepath.Join(".hooks", "run-go-tests.sh"), "all"); err != nil {
 		return fmt.Errorf("failed to run unit tests: %v", err)
 	}
 
+	fmt.Println("Running integration tests.")
+	if err := runIntegrationTests(); err != nil {
+		return fmt.Errorf("failed to run integration tests: %v", err)
+	}
+
+	return nil
+}
+
+func runIntegrationTests() error {
+	home, err := sys.GetHomeDir()
+	if err != nil {
+		return err
+	}
+	armoryTTPs := filepath.Join(home, ".ttpforge", "repos", "forgearmory", "ttps")
+	return findReadmeFiles(armoryTTPs)
+}
+
+func processLines(r io.Reader, language string) ([]string, error) {
+	scanner := bufio.NewScanner(r)
+	var lines, codeBlockLines []string
+	var inCodeBlock bool
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		inCodeBlock, codeBlockLines = handleLineInCodeBlock(strings.TrimSpace(line), line, inCodeBlock, language, codeBlockLines)
+
+		if !inCodeBlock {
+			lines = append(lines, codeBlockLines...)
+			codeBlockLines = codeBlockLines[:0]
+			if !strings.HasPrefix(line, "```") {
+				lines = append(lines, line)
+			}
+		}
+	}
+
+	if inCodeBlock {
+		codeBlockLines = append(codeBlockLines, "\t\t\t// ```")
+		lines = append(lines, codeBlockLines...)
+	}
+
+	return lines, scanner.Err()
+}
+
+func handleLineInCodeBlock(trimmedLine, line string, inCodeBlock bool, language string, codeBlockLines []string) (bool, []string) {
+	switch {
+	case strings.HasPrefix(trimmedLine, "```"+language):
+		if !inCodeBlock {
+			codeBlockLines = append(codeBlockLines, line)
+		}
+		return !inCodeBlock, codeBlockLines
+	case inCodeBlock:
+		codeBlockLines = append(codeBlockLines, line)
+	case strings.Contains(trimmedLine, "```"):
+		inCodeBlock = false
+	}
+	return inCodeBlock, codeBlockLines
+}
+
+func extractTTPForgeCommand(r io.Reader) ([]string, error) {
```

Comment:
Hey! can u add context about the approach ("parse out the ttpforge run command from readme and then run it") as code comments